server.go

```diff
@@ -2294,17 +2341,35 @@ func (s *server) ConnectToPeer(addr *lnwire.NetAddress, perm bool) error {
 	// connect to the target peer. If the we can't make the connection, or
 	// the crypto negotiation breaks down, then return an error to the
 	// caller.
+	errChan := make(chan error, 1)
+	s.connectToPeer(addr, errChan)
+
+	select {
+	case err := <-errChan:
+		return err
+	case <-s.quit:
```

Comment:
can also return `ErrServerShuttingDown` in this case

Reply:
Fixed.